plugins/chat/app/controllers/chat/api/channel_threads_controller.rb

```diff
@@ -80,15 +80,16 @@ def create
           membership:,
         )
       end
-      on_failed_policy(:threading_enabled_for_channel) { raise Discourse::NotFound }
-      on_failed_policy(:can_view_channel) { raise Discourse::InvalidAccess }
-      on_failed_step(:create_thread) do |step|
-        render json: failed_json.merge(errors: [step.error]), status: 422
-      end
       on_failure { render(json: failed_json, status: 422) }
       on_failed_contract do |contract|
         render(json: failed_json.merge(errors: contract.errors.full_messages), status: 400)
       end
+      on_model_not_found(:channel) { raise Discourse::NotFound }
+      on_failed_policy(:can_view_channel) { raise Discourse::InvalidAccess }
+      on_failed_policy(:threading_enabled_for_channel) { raise Discourse::NotFound }
+      on_model_errors(:thread) do |model|
```

Comment:
The previous `on_failed_step(:create_thread)` didn’t exist in the service, it has been replaced by a proper `model` step.

Reply:
Ah yes, I can see we had we had the `find_or_create_thread` step before, I wonder if that got changed at some point. Well spotted 👍